### ISSUE TITLE
Exploratory data analysis & preprocessing of training dataset

### ISSUE THREAD

**Issue by millerh1** [Bishop-Laboratory/autoPLIER-analysis#2]
**Goals**

1. Explore different approaches for normalizing the data.
2. Determine appropriate methods for filtering the dataset (e.g., removing samples with less than x million total read counts)
3. Exploring the structure of the dataset in relation to metadata labels 

**Questions answered**

1. What normalization strategy should we use?
2. What samples should be considered in the training set?
3. What features best explain the variance in the data? Are they biological (e.g., sex) or technical (e.g., # of reads) in nature?
4. Does clustering of the data produce a clear distinction between biologically-different samples (e.g., cancer vs normal, male vs female, brain vs bone, etc)

**Approach**

This analysis should be performed in a jupyter notebook (or RMarkdown). I would personally try the steps used to generate Figure 1A in the Cancers paper Methods (https://www.mdpi.com/2072-6694/12/4/948) and then optimize these if I was getting a reasonable result:

1. Filter out samples with < 10 million total read counts.
2. Apply the variance stabilizing transform (VST) to normalize the data.
3. Remove all but the top 10,000 most variable genes across samples (variable genes have the highest statistical variance of VST-transformed expression across samples -- in R, the rowVars() function calculates this).
4. Perform PCA on the resulting matrix -- use the elbow method to decide how many PCs to keep.
5. Perform clustering using louvain or leiden clustering -- and then calculate the UMAP embedding. 
6. Plot the resulting UMAP with tissue category annotations overlaid. 

These steps (and others) are done *mostly* automatically in the workflow for scRNA-Seq analysis. Try following the `Seurat` (R) or `scanpy` (Python) tutorials for more info.  

**Output**

The output should be:

1. A training dataset which is fully preprocessed and ready for classification.
2. Relevant figures and methods from answering the questions above. 

**Depends on**

#1 

However, there is a limited, older training set you can start with available here: 

Metadata: https://bishoplaboratory.slack.com/files/UJU46Q9PG/F01CC5F9XT9/training_data_v1_10_16_hm.csv
Expression data (needs to be wrangled): https://mssm-seq-matrix.s3.amazonaws.com/human_matrix.h5
